h/api/schemas.py

```diff
@@ -264,17 +192,50 @@ class CreateAnnotationSchema(object):
     """Validate the POSTed data of a create annotation request."""
 
     def __init__(self, request):
-        self.structure = AnnotationSchema(request)
+        self.structure = AnnotationSchema()
+        self.request = request
 
     def validate(self, data):
-        return self.structure.validate(data)
+        appstruct = self.structure.validate(data)
+
+        new_appstruct = {}
+
+        _remove_protected_fields(appstruct)
+
+        new_appstruct['userid'] = self.request.authenticated_userid
+        new_appstruct['target_uri'] = appstruct.pop('uri', u'')
```

Comment:
Hmm, we might want to not allow this specific case, because now we can introduce some sensible validations. For example, we [don't migrate](https://github.com/hypothesis/h/blob/master/scripts/migrate.py#L212-L213) any annotations with a missing or empty `uri` field